File: src/main/java/kor/toxicity/cutscenemaker/commands/SenderType.java
package kor.toxicity.cutscenemaker.commands;

import org.bukkit.command.CommandSender;
import org.bukkit.command.ConsoleCommandSender;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;

public enum SenderType {

    CONSOLE(ConsoleCommandSender.class),
    PLAYER(Player.class),
    ENTITY(Entity.class),
    ;

    public final Class<? extends CommandSender> sender;

    SenderType(Class<? extends CommandSender> s) {
        sender = s;
    }
}

File: plugin/src/main/java/kor/toxicity/cutscenemaker/command/SenderType.kt
package kor.toxicity.cutscenemaker.command

import org.bukkit.command.CommandSender
import org.bukkit.command.ConsoleCommandSender
import org.bukkit.entity.LivingEntity
import org.bukkit.entity.Player

enum class SenderType(
    val display: String,
    private val sender: Class<out CommandSender>
) {
    PLAYER("Player",Player::class.java),
    CONSOLE("Console",ConsoleCommandSender::class.java),
    LIVING_ENTITY("Entity",LivingEntity::class.java)
    ;
    fun accept(sender: Class<out CommandSender>) = this.sender.isAssignableFrom(sender)
}